teos/src/main.rs

```diff
@@ -186,14 +183,53 @@ async fn main() {
     // Load last known block from DB if found. Poll it from Bitcoind otherwise.
     let last_known_block = dbm.lock().unwrap().load_last_known_block();
     let tip = if let Some(block_hash) = last_known_block {
-        derefed
+        let mut last_known_header = derefed
             .get_header(&block_hash, None)
             .await
             .unwrap()
             .validate(block_hash)
-            .unwrap()
+            .unwrap();
+
+        log::info!(
+            "Last known block: {} (height: {})",
+            last_known_header.header.block_hash(),
+            last_known_header.height
+        );
+
+        // If we are running in pruned mode some data may be missing (if we happen to have been offline for a while)
+        if let Some(prune_height) = rpc.get_blockchain_info().unwrap().prune_height {
+            if last_known_header.height - IRREVOCABLY_RESOLVED < prune_height as u32 {
```

Comment:
This has an off-by-one error :joy:, should be `last_known_header.height - IRREVOCABLY_RESOLVED + 1 < prune_height as u32`.

Let's say we last witnessed block 3000, then we will want to load blocks 2901-3000 (100 blocks). `prune_height` indicates the lowest block ***we have*** stored, so if `prune_height` is 2901 we should be fine, but `last_known_header.height - IRREVOCABLY_RESOLVED` will be 2900, which would make the check fail.

Reply:
You're right